hydra-node/src/Hydra/Crypto.hs

```diff
@@ -180,9 +180,8 @@ generateSigningKey :: ByteString -> SigningKey HydraKey
 generateSigningKey seed =
   HydraSigningKey . genKeyDSIGN $ mkSeedFromBytes padded
  where
-  needed = fromIntegral $ seedSizeDSIGN (Proxy :: Proxy Ed25519DSIGN)
-  provided = BS.length seed
-  padded = seed <> BS.pack (replicate (needed - provided) 0)
+  hashOfSeed = digest (Proxy :: Proxy SHA256) seed
+  padded = hashOfSeed <> seed
```

Comment:
The derived key can only be the hash. No need to suffix the seed. Also, the variable name `padded` does not make sense anymore. Just use `hashOfSeed` instead of it.